### DISCUSSION TITLE
Avoid Trailing Whitespaces

### DISCUSSION THREAD

**Discussion by christianbraeunlich** [StefanMaron/BusinessCentral.LinterCop#173] [New Rule]
I'd like to avoid trailing whitespaces.

Inspired by:
- https://www.python.org/dev/peps/pep-0008/#other-recommendations
- https://www.flake8rules.com/rules/W291.html

**Comment by rvanbekkum** [comment]
It's not a rule, but the code cleanup command in the AL Code Outline/AZ AL Dev Tools has an option to trim trailing spaces: 🙂https://github.com/anzwdev/al-code-outline#code-transformation-commands

**Reply by christianbraeunlich** [reply]
Thanks for your reply. Yes indeed `alOutline.codeCleanupActions` works perfectly fine. However I still like the idea of having a pipeline step to make sure at a **_centralized_** point that this rule has been checked. This way we can be sure that no trailing whitespaces have been inserted accidentally. You can already tell that I'm a bit too Pythonic biased here...so I would like to discuss whether this could be a good rule or if there might be other/better tools for this.
